detectable/src/test/java/com/synopsys/integration/detectable/detectables/npm/cli/unit/NpmDependencyTypeFilterTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * detectable
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.detectable.detectables.npm.cli.unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.synopsys.integration.detectable.detectables.npm.cli.parse.NpmDependencyTypeFilter;
+import com.synopsys.integration.util.Stringable;
+
+class NpmDependencyTypeFilterTest {
+
+    @ParameterizedTest()
+    @MethodSource("generateTest")
+    void includeDependencies(TestParameter testParameter) {
+        Set<String> devDependencies = Collections.singleton("test-dev");
+        Set<String> peerDependencies = Collections.singleton("test-peer");
+
+        NpmDependencyTypeFilter npmDependencyTypeFilter = new NpmDependencyTypeFilter(devDependencies, peerDependencies, testParameter.includeDevDependencies, testParameter.includePeerDependencies);
+        String dependencyName = "test";
+        if (testParameter.isDevDependency) {
+            dependencyName = "test-dev";
+        } else if (testParameter.isPeerDependency) {
+            dependencyName = "test-peer";
+        }
+        boolean actual = npmDependencyTypeFilter.shouldInclude(dependencyName, testParameter.isRootDependency);
+        assertEquals(testParameter.shouldInclude, actual, "TestParameters: " + testParameter);
+    }
+
+    static Stream<TestParameter> generateTest() {
+        return Stream.<TestParameter>builder()
```

Comment:
I am not a big fan of this, it's hard to tell what is meaningful.. Could it at least split to a should include and should not include - each with their own parameters? That way at least the assertion isn't a parameter.